File: C#/1004. Max Consecutive Ones III.cs
/*
 * Link: https://leetcode.com/problems/max-consecutive-ones-iii/
 * Author: Dung Nguyen Tien (Chris)
*/

public class Solution {
    public int LongestOnes(int[] A, int K) {
        int begin = 0, end = 0;
        int count = 0, maxLen = 0;
        
        while(end < A.Length) {
            if (A[end++] == 0) {
                count++;
            }
            
            while(count > K) {
                if (A[begin++] == 0) count--;
            }
            
            maxLen = Math.Max(maxLen, end - begin);
        }
        
        return maxLen;
    }
}

File: Python/1004. Max Consecutive Ones III.py
class Solution(object):
    def longestOnes(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        start, end, count, maxLen = 0, 0, 0, 0
        while end < len(nums):
            if nums[end] == 0:
                count += 1
            end += 1
            while(count > k):
                if nums[start] == 0:
                    count -= 1
                start += 1
            maxLen = max(maxLen, end - start)
        return maxLen